server/src/test/integration/org/apache/cassandra/sidecar/cluster/locator/CqlSessionProviderIntegrationTest.java

```diff
@@ -0,0 +1,163 @@
+                                                 .set("client_encryption_options.truststore_password", truststorePassword));
+        });
+        waitForSchemaReady(30, TimeUnit.SECONDS);
+        insertIdentityRole(ADMIN_IDENTITY, "cassandra");
+        sidecarTestContext.setSslConfiguration(sslConfigWithKeystoreTruststore());
+        retrieveClientStats(context, "cassandra", true);
+    }
+
+    private SslConfiguration sslConfigWithTruststore()
+    {
+        return SslConfigurationImpl.builder()
+                                   .enabled(true)
+                                   .truststore(new KeyStoreConfigurationImpl(truststorePath.toAbsolutePath().toString(), truststorePassword, "PKCS12"))
+                                   .build();
+    }
+
+    private SslConfiguration sslConfigWithKeystoreTruststore()
+    {
+        return SslConfigurationImpl.builder()
+                                   .enabled(true)
+                                   .keystore(new KeyStoreConfigurationImpl(clientKeystorePath.toAbsolutePath().toString(), clientKeystorePassword, "PKCS12"))
+                                   .truststore(new KeyStoreConfigurationImpl(truststorePath.toAbsolutePath().toString(), truststorePassword, "PKCS12"))
+                                   .build();
+    }
+
+    private void retrieveClientStats(VertxTestContext context, String expectedUsername, boolean checkSsl) throws Exception
+    {
+        String testRoute = "/api/v1/cassandra/stats/connected-clients?summary=false";
+        WebClient client = mTLSClient();
+        client.get(server.actualPort(), "127.0.0.1", testRoute)
+              .send(context.succeeding(response -> context.verify(() -> {
+                  ConnectedClientStatsResponse clientStatsResponse = response.bodyAsJson(ConnectedClientStatsResponse.class);
+                  assertThat(clientStatsResponse).isNotNull();
+
+                  for (ClientConnectionEntry entry : clientStatsResponse.clientConnections())
+                  {
+                      assertThat(entry.username()).isEqualTo(expectedUsername);
+                      if (checkSsl && entry.sslEnabled())
+                      {
+                          // We expect some connections to be non-SSL (i.e. for identity setup)
+                          // and some connections to be SSL (Sidecar connecting to the cluster)
+                          // so from the list of client connections we should see at least
+                          // two (regular+control) connections.
+                          context.completeNow();
+                          return;
+                      }
+                  }
+                  if (checkSsl)
+                  {
+                      context.failNow("Did not see any SSL connection");
+                      return;
+                  }
+                  context.completeNow();
+              })));
```

Comment:
`context.completeNow();` has to be outside of the `verify` block. Otherwise, any assertion failure causes the context not gets completed. 
Besides that, I moved the code around a bit to hopefully simplify the assertion. 

```suggestion
              .send(context.succeeding(response -> {
                  context.verify(() -> {
                      ConnectedClientStatsResponse clientStatsResponse = response.bodyAsJson(ConnectedClientStatsResponse.class);
                      assertThat(clientStatsResponse).isNotNull();

                      boolean seeSslConnection = false;
                      for (ClientConnectionEntry entry : clientStatsResponse.clientConnections())
                      {
                          assertThat(entry.username()).isEqualTo(expectedUsername);
                          if (checkSsl && entry.sslEnabled())
                          {
                              seeSslConnection = true;
                              break;
                          }
                      }
                      // We expect some connections to be non-SSL (i.e. for identity setup)
                      // and some connections to be SSL (Sidecar connecting to the cluster)
                      // so from the list of client connections we should see at least
                      // two (regular+control) connections.
                      assertThat(seeSslConnection)
                      .describedAs("Did not see any SSL connection")
                      .isTrue();
                  });
                  context.completeNow();
              }));
```